misk/src/main/kotlin/misk/client/ClientInvocationHandler.kt

```diff
@@ -39,6 +39,7 @@ internal class ClientInvocationHandler(
   private val clientName: String,
   retrofit: Retrofit,
   okHttpTemplate: OkHttpClient,
+  clientInterceptorFactories: Provider<List<ClientInterceptors.Factory>>,
```

Comment:
Is it enough to do in this class only? I think I have seen places using HttpClientFactory directly, without doing anything with ClientInvocationHandler. Maybe it's better to inject it directly there?